src/Chainweb/Version.hs

```diff
@@ -420,19 +421,24 @@ instance NFData PactUpgrade where
 pact4Upgrade :: [Pact4.Transaction] -> PactUpgrade
 pact4Upgrade txs = Pact4Upgrade txs False
 
+data TxIdxInBlock = TxBlockIdx Word
+    deriving stock (Eq, Ord, Show, Generic)
+    deriving anyclass (Hashable, NFData)
+
+makePrisms ''TxIdxInBlock
+
 -- The type of quirks, i.e. special validation behaviors that are in some
 -- sense one-offs which can't be expressed as upgrade transactions and must be
 -- preserved.
 data VersionQuirks = VersionQuirks
-    { _quirkGasFees :: !(HashMap RequestKey Gas)
-    -- ^ Note: only works for user txs in blocks right now.
+    { _quirkGasFees :: !(ChainMap (HashMap (BlockHeight, TxIdxInBlock) Gas))
```

Comment:
Why the migration to `(BlockHeight, TxIdxInBlock)` instead of request key? I don't understand the necessity/benefit of this change.